core/domain/search_services.py

```diff
@@ -0,0 +1,301 @@
+            '+' or a '-' character indicating whether to sort in ascending or
+            descending order respectively. This character should be followed by
+            a field name to sort on. When this is None, results are based on
+            'rank'. get_search_rank_from_exp_summary to see how
+            rank is determined.
+        cursor: str or None. A cursor, used to get the next page of results. If
+            there are more documents that match the query than 'limit', this
+            function will return a cursor to get the next page.
+
+    Returns:
+        tuple. A 2-tuple consisting of:
+            - list(str). A list of exploration ids that match the query.
+            - str or None. A cursor if there are more matching explorations to
+              fetch, None otherwise. If a cursor is returned, it will be a
+              web-safe string that can be used in URLs.
+    """
+    return search_services.search(
+        query, SEARCH_INDEX_EXPLORATIONS, cursor, limit, sort, ids_only=True)
+
+
+def delete_explorations_from_search_index(exploration_ids):
+    """Deletes the documents corresponding to these exploration_ids from the
+    search index.
+
+    Args:
+        exploration_ids: list(str). A list of exploration ids whose
+            documents are to be deleted from the search index.
+    """
+    search_services.delete_documents_from_index(
+        exploration_ids, SEARCH_INDEX_EXPLORATIONS)
+
+
+def patch_exploration_search_document(exp_id, update):
+    """Patches an exploration's current search document, with the values
+    from the 'update' dictionary.
+
+    Args:
+        exp_id: str. The id of the exploration to be patched.
+        update: dict. Key-value pairs to patch the exploration's search
+            document with.
+    """
+    doc = search_services.get_document_from_index(
+        exp_id, SEARCH_INDEX_EXPLORATIONS)
+    doc.update(update)
+    search_services.add_documents_to_index([doc], SEARCH_INDEX_EXPLORATIONS)
+
+
+def clear_exploration_search_index():
+    """WARNING: This runs in-request, and may therefore fail if there are too
+    many entries in the index.
+    """
+    search_services.clear_index(SEARCH_INDEX_EXPLORATIONS)
+
+def search_collections(query, limit, sort=None, cursor=None):
+    """Searches through the available collections.
+
+    Args:
+        query_string: str. the query string to search for.
+        sort: str. This indicates how to sort results. This should be a string
+            of space separated values. Each value should start with a '+' or a
+            '-' character indicating whether to sort in ascending or descending
+            order respectively. This character should be followed by a field
+            name to sort on. When this is None, results are returned based on
+            their ranking (which is currently set to the same default value
+            for all collections).
+        limit: int. the maximum number of results to return.
```

Comment:
Shift it above the sort. Follow the order in arguments.

Reply:
Done.